packages/block-library/src/button/style.scss

```diff
@@ -48,7 +48,7 @@ $blocks-button__height: 56px;
 	border-radius: 0 !important;
 }
 
-.wp-block-button.is-style-outline,
```

Comment:
:wave: Riad, in this [forum post](https://wordpress.org/support/topic/issue-with-wp-buttons-having-a-transparent-background-post-5-5-update/#post-13268135) a user reported that in WordPress 5.5, buttons with outline style and a preset color background didn't have the background color set.

My first instinct is to think that was how it was expected to work. However:

- In 5.4 it worked differently, the backgrounds were also applied to buttons with outline style (both editor and front).
- In 5.5, the editor and front reflect different values at the moment. If this is working as expected in the front, we should make it the same in the editor.

The button markup changed a bit during this cycle, so it's difficult to point to a specific PR that introduced the regression. This one is as good a candidate as any, as this PR increased the selector specificity to 030, while in WordPress 5.4 it was 020 (`.is-style-outline .wp-block-button__link`).

I've also tested in other themes and the results are somewhat inconsistent:

- 2019: it works as before (uses background even with outline style) because the theme selector overrides the core selector by virtue of being loaded after (they have the same specificity).
- 2017: it doesn't show the background when the outline style is selected (theme selector has lower specificity than core's).

I'm inclined to think this is a regression we need to fix. What are your thoughts on this @youknowriad ? cc @kjellr

Reply:
Thanks for digging here. It's definitely not an intended change so I agree with considering this a regression.

Reply:
Got a potential fix at https://github.com/WordPress/gutenberg/pull/24599

@ellatrix do we have an incoming 5.5.1? This issue may be a good one to backport. If the fix is approved, should I backport it to the `wp/5.5`? Do I need to do anything else?

Reply:
Yeah, I agree that it sounds like a regression. Thanks for flagging that!